feat : Add Article expiration reminder support

gclm opened the pull request:
效果如下所示：
![image](https://user-images.githubusercontent.com/27618687/136198085-6a822a38-e446-462c-a41f-1e61772aa7db.png)

gclm committed:
feat :Add Article expiration reminder support

gclm committed:
feat: Add Article expiration reminder support

HEIGE-PCloud commented:
我想做的一些改动

- 将 `params.outdatedInfoWarning` 放到 `params.page.outdatedInfoWarning` 下面。这将允许每个页面单独开启/关闭/配置这个警告。
- 将 warning 放到开头，而非文章末尾。

gclm committed:
fix: Adjust warning position

gclm commented:
> 我想做的一些改动
> 
> * 将 `params.outdatedInfoWarning` 放到 `params.page.outdatedInfoWarning` 下面。这将允许每个页面单独开启/关闭/配置这个警告.

这个我已经做过了。**outdatedInfoWarning: false** 可以添加在每篇文章的文件头里来控制是否启用该功能：

> * 将 warning 放到开头，而非文章末尾。

警告位置我已经修改过了。

HEIGE-PCloud committed:
chore: clean up code and add documentation

- Rename outdatedInfoWarning to outdatedArticleReminder
- Rename hint to reminder
- Rename warn to warning
- Move outdatedArticleReminder from params to params.pages
- Add documentation for example config
- Add documentation for front matter
- Update translation for en and zh-CN
- clean up code for outdatedArticleReminder.html

HEIGE-PCloud commented:
我做了一些 code clean up，加了一些文档。麻烦你再看一下，如果没有问题就可以合并了 👍

HEIGE-PCloud committed:
Merge branch 'main' into dev-gclm

gclm commented:
我已经测试过了没问题。 @HEIGE-PCloud

gclm committed:
fix: update default reminder time

CXwudi commented on exampleSite/config.toml:
```diff
@@ -629,6 +629,16 @@ enableEmoji = true
     pageStyle = "wide"
+    # Display a message at the beginning of an article to warn the reader that its content might be outdated.
+    # 在文章开头显示提示信息，提醒读者文章内容可能过时。
+    [params.pages.outdatedArticleReminder]
```
想问一下, key值为`params.pages`而不是`params.page`有什么原因吗?
以及要是能改进一下indentation就好了

HEIGE-PCloud replied:
抱歉... 已修改...
